core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1158,12 +1158,18 @@ public int getNumberOfPinnedFiles() {
   public void delete(AlluxioURI path, DeleteOptions options) throws IOException,
       FileDoesNotExistException, DirectoryNotEmptyException, InvalidPathException,
       AccessControlException {
+    List<Inode<?>> deletedInodes;
     Metrics.DELETE_PATHS_OPS.inc();
     try (JournalContext journalContext = createJournalContext();
          LockedInodePath inodePath = mInodeTree.lockFullInodePath(path, InodeTree.LockMode.WRITE)) {
       mPermissionChecker.checkParentPermission(Mode.Bits.WRITE, inodePath);
       mMountTable.checkUnderWritableMountPoint(path);
-      deleteAndJournal(inodePath, options, journalContext);
+      deletedInodes = deleteAndJournal(inodePath, options, journalContext);
+    }
+    for (Inode<?> inode : deletedInodes) {
```

Comment:
factored out

I switched to batching and expanded the scope of `journalContext` in `BlockMaster#removeBlocks` to take advantage of it. Does the change of journalContext scope make sense?